derive_client/_bridge/client.py

```diff
@@ -528,78 +529,64 @@ def poll_bridge_progress(self, tx_result: BridgeTxResult) -> BridgeTxResult:
+            # 1. TimeoutError as exception during source_tx.tx_receipt
+            if not tx_result.source_tx.tx_receipt:
+                msg = "⏳ Checking source chain [%s] tx receipt for %s"
+                self.logger.info(msg, tx_result.source_chain.name, tx_result.source_tx.tx_hash)
                 tx_result.source_tx.tx_receipt = wait_for_tx_receipt(
                     w3=context.source_w3, tx_hash=tx_result.source_tx.tx_hash
                 )
-            except TimeoutError as e:
-                tx_result.source_tx.exception = e
-                return tx_result
 
-        # 2. target_tx is None (i.e. TimeoutError when waiting for event log on target chain)
-        if not tx_result.target_tx:
-            try:
+            # 2. target_tx is None (i.e. TimeoutError when waiting for event log on target chain)
+            if not tx_result.target_tx:
                 event_log = fetch_event(tx_result, context)
                 tx_result.target_tx = TxResult(event_log["transactionHash"].to_0x_hex())
-            except TimeoutError:
-                return tx_result
 
-        # 3. Timeout waiting for target_tx.tx_receipt
-        if not tx_result.target_tx.tx_receipt:
-            self.logger.info(
-                f"⏳ Checking target chain [{tx_result.target_chain.name}] tx receipt for {tx_result.target_tx.tx_hash}"
-            )
-            tx_result.target_tx.exception = None
-            try:
+            # 3. TimeoutError waiting for target_tx.tx_receipt
+            if not tx_result.target_tx.tx_receipt:
+                msg = "⏳ Checking target chain [%s] tx receipt for %s"
+                self.logger.info(msg, tx_result.target_chain.name, tx_result.target_tx.tx_hash)
                 tx_result.target_tx.tx_receipt = wait_for_tx_receipt(
                     w3=context.target_w3, tx_hash=tx_result.target_tx.tx_hash
                 )
-            except TimeoutError as e:
-                tx_result.target_tx.exception = e
 
         return tx_result
 
-    def _ensure_derive_eth_balance(self, tx:dict[str, str]):
+    def _ensure_derive_eth_balance(self, tx: dict[str, str]):
         """Ensure that the Derive EOA wallet has sufficient ETH balance for gas."""
-        balance_of_owner = self.derive_w3.eth.get_balance(self.owner)
-        required_gas = tx['maxFeePerGas'] * tx['gas']
-        if balance_of_owner < required_gas + DEFAULT_GAS_FUNDING_AMOUNT:
-            self.logger.info(f"Funding Derive EOA wallet with {DEFAULT_GAS_FUNDING_AMOUNT} ETH")
-            self.bridge_mainnet_eth_to_derive(DEFAULT_GAS_FUNDING_AMOUNT)
-
-    def bridge_mainnet_eth_to_derive(self, amount: int) -> TxResult:
-        """
-        Prepares, signs, and sends a transaction to bridge ETH from mainnet to Derive.
-        This is the "socket superbridge" method; not required when using the withdraw wrapper.
-        """
+        self.logger.info(f"Funding Derive EOA wallet with {DEFAULT_GAS_FUNDING_AMOUNT} ETH")
```

Comment:
`DEFAULT_GAS_FUNDING_AMOUNT` is fine as a baseline, but funding logic should account for:
1. Lower bound: compare against actual estimated requirement and override if higher
2. Upper bound: cap funding to avoid overexposing (i.e. gas spikes)
3. Buffer: fund enough for multiple L2 txs, to avoid repeated bridging